barman/postgres.py

```diff
@@ -1325,9 +1325,12 @@ def switch_wal(self):
         """
         try:
             conn = self.connect()
-            # Requires superuser privilege
+            # Requires superuser privilege or specific writes
             if not self.has_backup_privileges:
-                raise BackupFunctionsAccessRequired()
+                msg = "{} user does not have enough privileges to run switch-wal (see documentation).".format(
+                    self.application_name
```

Comment:
This should be the username, not the application name.

Reply:
Indeed, I though the username was the application name. 
changed it fo `self.conn_parameters.get("user")`